src/mesh/voxel_cloud.rs

```diff
@@ -214,6 +213,211 @@ impl ScalarField {
+                    );
+                    *voxel = source_scalar_field
+                        .value_at_absolute_voxel_coordinate(&absolute_coordinate_source);
+                }
+
+                // FIXME: @Optimization In some cases it might be not easy to
+                // determine the minimal needed bounding box due to the
+                // differences in the voxel sizes. Leaving the scalar field
+                // bigger should be also considered.
+                target_scalar_field.shrink_to_fit(volume_value_range);
+
+                target_scalar_field
+            })
+    }
+
+    /// Creates a new scalar field from another scalar field transformed
+    /// (scaled, rotated, moved) in a cartesian space.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the voxel dimension is below or equal to zero.
+    pub fn from_scalar_field_transformed<U>(
+        source_scalar_field: &ScalarField,
+        volume_value_range: &U,
+        voxel_dimensions: &Vector3<f32>,
+        cartesian_translation: &Vector3<f32>,
+        rotation: &Rotation3<f32>,
+        scale_factor: &Vector3<f32>,
+    ) -> Option<Self>
+    where
+        U: RangeBounds<f32>,
+    {
+        let euler_angles = rotation.euler_angles();
+        // If the transformation is identity
+        if approx::relative_eq!(cartesian_translation, &Vector3::zeros())
+            && approx::relative_eq!(euler_angles.0, 0.0)
+            && approx::relative_eq!(euler_angles.1, 0.0)
+            && approx::relative_eq!(euler_angles.2, 0.0)
+            && approx::relative_eq!(scale_factor, &Vector3::new(1.0, 1.0, 1.0))
+        {
+            // and the voxel dimensions don't change
+            if approx::relative_eq!(voxel_dimensions, &source_scalar_field.voxel_dimensions) {
+                // return identical copy of self
+                return Some(ScalarField {
+                    block_start: source_scalar_field.block_start,
+                    block_dimensions: source_scalar_field.block_dimensions,
+                    voxel_dimensions: source_scalar_field.voxel_dimensions,
+                    voxels: source_scalar_field.voxels.to_vec(),
+                });
+            } else {
+                // or resample to new voxel dimensions and return.
+                return ScalarField::from_scalar_field(
+                    source_scalar_field,
+                    volume_value_range,
+                    voxel_dimensions,
+                );
+            }
+        }
+
+        assert!(
+            voxel_dimensions.x > 0.0 && voxel_dimensions.y > 0.0 && voxel_dimensions.z > 0.0,
+            "One mor more voxel dimensions is below or equal to zero"
+        );
+
+        // Make a bounding box of the source scalar field's mesh volume. This
+        // will be the volume to be scanned for any voxels.
+        if let Some(source_sf_bounding_box) =
+            source_scalar_field.bounding_box_mesh_volume_cartesian_space(volume_value_range)
```

Comment:
I'll look into it. Th reason for taking the mesh volume bounding box is the possible difference in voxel dimensions. 
If this causes some offset, the error must be somewhere in the math rather than in the usage of mesh volume.

Reply:
You are right - we still should be thinking that (0,0,0) in cartesian coordinates is the center of the voxel. Now to just find the meth error...

Reply:
Now when I gave ti second, third and fourth thought, I'm not sure anymore if the meshing volume actually matters. In the current iteration, I'm trimming the bounding box to the voxel centers. I'm actually not sure if there is a correct answer.

Reply:
> Now when I gave ti second, third and fourth thought, I'm not sure anymore if the meshing volume actually matters.

Could you explain?

> In the current iteration, I'm trimming the bounding box to the voxel centers.

Won't this be less symmetrical? Most geometry will be centered around origin. Don't we want this to give good results (not shifting the geometry) for these cases? 

> I'm actually not sure if there is a correct answer.

It is a bit of an arbitrary decision, but having voxels in voxel space be centers of voxels in catesian space (instead of front-bottom-left corners) seems like a good decision.